packages/cart/src/models/adjustment-line.ts

```diff
@@ -0,0 +1,41 @@
+import { DAL } from "@medusajs/types"
+import { OptionalProps, PrimaryKey, Property } from "@mikro-orm/core"
+
+type OptionalAdjustmentLineProps = DAL.EntityDateColumns // TODO: To be revisited when more clear
+
+export default class AdjustmentLine {
+  [OptionalProps]: OptionalAdjustmentLineProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({ columnType: "text", nullable: true })
+  description?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  promotion_id?: string | null
+
+  @Property({ columnType: "text" })
+  code: string
+
+  @Property({ columnType: "numeric" })
+  amount: number
```

Comment:
Yes, nice catch.

Reply:
We need a util to be used for money amounts everywhere to normalize this. Probably a good time to sync and update pricing, promotion and cart.

Reply:
Yeah, would make sense.

Were you thinking something along the lines of what [this guy proposes](https://github.com/typeorm/typeorm/issues/873#issuecomment-424643086), but instead of `parseFloat` use bignumber.js to return a BigNumber when reading the column?

Reply:
yes. we could install bignumber.js on our utils and use it from there.
however that will impact the calculations. but we can do it in two steps.
first normalizing all the db + serialize but returning the BigNumber .toNumber(), and next step use the methods to perform the calculations where it is needed.

Reply:
This sounds like a reasonable approach to me.

@pKorsholm @riqwan @adrien2p, what do you think?

Reply:
That sounds good to me 👌

Reply:
yes, that sounds good. Since bignumberjs also works with strings, this would amount to `amount: string`? or are we still serializing it back to a number.

Reply:
yes the property can be kept as number, it doesn't prevent us from using float numbers for example. But it would be more about the serizliation and deserialization related no?

Reply:
just to summarize the idea from carlos
- step1: everything number with current ser/deser
- step2: include bignumbers or similar as the ser/deser to number
- step3: change number usage into all calculations